Fail to find RTCOptions in 3.1.2 UNITY_EDITOR_OSX (3.1.1 works)

Library/PackageCache/com.playeveryware.eos@c13536130d/Runtime/Core/PlatformSpecifics.cs(91,55): error CS1061: 'Options' does not contain a definition for 'RTCOptions' and no accessible extension method 'RTCOptions' accepting a first argument of type 'Options' could be found (are you missing a using directive or an assembly reference?)

--- a/Assets/Plugins/Source/Core/EOSPackageInfo.cs
+++ b/Assets/Plugins/Source/Core/EOSPackageInfo.cs
@@ -35,7 +35,7 @@ namespace PlayEveryWare.EpicOnlineServices
          * not involve editing source code files.
          */
 
-        public const string Version = "3.1.2";
+        public const string Version = "3.1.3";
 
         public const string PackageName = "com.playeveryware.eos";
     }

--- a/Assets/Plugins/Source/Editor/Utility/BuildUtility.cs
+++ b/Assets/Plugins/Source/Editor/Utility/BuildUtility.cs
@@ -70,25 +70,17 @@ namespace PlayEveryWare.EpicOnlineServices.Build
         /// <returns>True if the plugin is deployed as UPM, false otherwise.</returns>
         private static bool IsDeployedAsUPM()
         {
-            try
+            string packagePathname = Path.GetFullPath(Path.Combine("Packages", EOSPackageInfo.PackageName));
+
+            if (Directory.Exists(packagePathname))
             {
-                string manifestJson = FileUtility.ReadAllText(
-                    Path.Combine(FileUtility.GetProjectPath(), "Packages", "manifest.json")
-                );
-
-                Manifest manifest = JsonUtility.FromJson<Manifest>(manifestJson);
-
-                if (manifest != null && manifest.dependencies.ContainsKey(EOSPackageInfo.PackageName))
-                {
-                    return true;
-                }
+                Debug.Log("Deployed via UPM");
+                return true;
             }
-            catch (Exception e)
+            else
             {
-                Debug.LogWarning($"There was a problem determining if deployed via UPM: \"{e.Message}\".");
+                return false;
             }
-
-            return false;
         }
 
         /// <summary>

--- a/Assets/Plugins/macOS/Core/MacOSPlatformSpecifics.cs
+++ b/Assets/Plugins/macOS/Core/MacOSPlatformSpecifics.cs
@@ -32,7 +32,6 @@ using UnityEngine.Scripting;
 namespace PlayEveryWare.EpicOnlineServices 
 {
     using Epic.OnlineServices.Platform;
-    using Options = Epic.OnlineServices.IntegratedPlatform.Options;
 
     public class EOSCreateOptions
     {

--- a/etc/PackageTemplate/CHANGELOG.md
+++ b/etc/PackageTemplate/CHANGELOG.md
@@ -2,6 +2,12 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+# [3.1.3] - 2024-04-29
+
+### Fixed
+fix(build,upm) : fix check if deployed from upm or repo
+fix(mac) : remove incorrect create option type on mac
+
 # [3.1.2] - 2024-04-25
 
 ### Removed

--- a/etc/PackageTemplate/CHANGELOG.md
+++ b/etc/PackageTemplate/CHANGELOG.md
@@ -2,6 +2,28 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+# [3.1.2] - 2024-04-25
+
+### Removed
+chore: Remove EOSCreateOptions and EOSInitializeOptions files in favor of defining them within individual PlatformSpecifics files.
+chore: Remove commented out code.
+
+### Fixed
+fix(build,upm) : Compile fixes when running in a upm
+fix: Re-order platform if-defs.
+fix: Add UpdateNetworkStatus to both interface and abstract implementing class.
+fix: Instead of explicitly instantiating config editors, use reflection to determine all platform config editors that exist, and create an instance for each of them in EOSSettingsWindow.
+fix: Restore if-def block around IOSPlatformSpecifics.
+fix: Return AndroidPlatformSpecifics to prior class name, and correct the config file indicated in .
+fix: Allow platform-specifics implementing classes to define the type of config and library extension for the platform.
+fix: Remove IEOSCreateOptions helper methods and interface, replacing with implementation.
+Fix name of AndroidPlatformSpecifics
+fix: Update version number string in EOSPackageInfo.
+fix: Correct unity version listing in package.json file.
+fix: Remove dependency on Newtonsoft.Json.
+fix: Add log error to catching exception when creating builder, as well as throw build failed exception.
+
+
 # [3.1.1] - 2024-04-15
 
 ### Fixed

--- a/etc/PackageTemplate/package.json
+++ b/etc/PackageTemplate/package.json
@@ -1,6 +1,6 @@
 {
     "name": "com.playeveryware.eos",
-    "version": "3.1.2",
+    "version": "3.1.3",
     "unity": "2021.3",
     "unityRelease": "8f1",
     "displayName": "Epic Online Services Plugin for Unity",
@@ -31,6 +31,6 @@
         }
     ],
     "com_playeveryware": {
-        "git_build_sha": "14ec6f0a30fc1806089326ff117022d242c79a3d"
+        "git_build_sha": "86f410958fb596b5a42c05419577c1ffc70cffba"
     }
 }

--- a/etc/PackageTemplate/package.json
+++ b/etc/PackageTemplate/package.json
@@ -1,6 +1,6 @@
 {
     "name": "com.playeveryware.eos",
-    "version": "3.1.1",
+    "version": "3.1.2",
     "unity": "2021.3",
     "unityRelease": "8f1",
     "displayName": "Epic Online Services Plugin for Unity",
@@ -31,6 +31,6 @@
         }
     ],
     "com_playeveryware": {
-        "git_build_sha": "31b5540304ff074d7ef01f9377dafcd8fc5b1dc4"
+        "git_build_sha": "14ec6f0a30fc1806089326ff117022d242c79a3d"
     }
 }

